apps/ensadmin-next/src/app/ponder-client-api/page.tsx

```diff
@@ -1,13 +1,49 @@
-import { PageShell } from "@/components/page-shell";
-import { ExamplesPage } from "@/components/ponder-client/examples-page";
 import { Suspense } from "react";
 
-export default function PonderClientPage() {
+import { preferredEnsNodeUrl } from "@/lib/env";
+import { PonderClientContent } from "./examples-content";
+import { Provider as PonderClientProvider } from "./provider";
+
+type PageProps = {
+  searchParams: Promise<{
+    [key: string]: string | string[] | undefined;
+  }>;
+};
+
+export default async function PonderClientPage({ searchParams }: PageProps) {
+  const { ensnode = preferredEnsNodeUrl() } = await searchParams;
+
+  const baseUrl = Array.isArray(ensnode)
+    ? ensnode[0]
+    : typeof ensnode === "string"
+      ? ensnode
+      : preferredEnsNodeUrl();
+
+  const url = new URL(`/subgraph`, baseUrl).toString();
+
+  return (
+    <Suspense fallback={<Loading />}>
+      <PonderClientProvider url={url}>
```

Comment:
I love how you moved the `PonderClientProvider` down the components tree. Previously, it'd require almost the whole tree to be made of client components, which was not good for page transitions smoothness.